### ISSUE TITLE
Restart button for inline animations

### ISSUE THREAD

**Issue by jo-pol** [d-bl/GroundForge#174]
## The problem
Repeating animation can have disadvantages
* distracting when reading the text around it
* not clear what is the starting point
* even when adding some fade-in fade-out as a visual mark, the user might have to wait too long for a fresh cycle

An animation that doesn't repeat also has disadvantages
* before the user knows something interesting is happening, (s) missed most of it
* the user may need to scroll to reach the image, in that case the animation will be finished completely before it gets into view

A button to restart a not repeating animation would fix the last problems. 

## A workaround:
* An image that gets the attention on the subject
* Some overlay suggesting there is something to start and that gives an indication about the duration

Example : https://d-bl.github.io/GroundForge/help/Change-tiles
* disadvantage of the video control bar: it suggests you can pause, which you can't

## task
Implement [css](https://github.com/d-bl/GroundForge/blob/master/docs/css/tiles.css)/javascript to (re)start an inline animated image that works on most browsers used by our audience.
A quick google resulted in various solutions, each with pro's and con's and different (lack of) browser support.

**Comment by Cowboynoon** [comment]
I think your assumptions about general users' ability to understand and navigate moving media is incorrect - the gif is not too long, and the sequence logical and clear. (Gifs have been part of the internet for a very long time and the assumption is that they should repeat rather than not, hence my next suggestion:)

I suggest keeping it looped because a still image (at the end of a cycle) can be confused with a jpeg and the additional clicks to refresh the animation is extra effort and dependent on reading instructions. 

The following notes are off-topic in this thread but I'm not sure where to comment:

As for the wording of the pattern arrange panel's help page, may I suggest "offset pattern" instead of "checkerboard" which is probably more correct for this arrangement anyway (I was not aware of the term's history here as referring to the simple repeat layout).

Just from a teaching perspective, I think it is important to settle on a name for the buttons for consistency (my checkerboard is someone else's bathroom tile" as we've established...) The terms "simple" "brick" and "offset" seem logical and clear.

I find this line a little problematic: "Any textbook will explain the bottom corners of the unit are the top corners for the next row. "We saw this happen when moving the tiles in place. Make these stitches faint with a dash. Having these stitches up to now might help to check the tiles arrived at their target: they blend"

-The textbook reference is vague and expressing a common sense observation (in a slightly condescending tone implied by starting the sentence with "any".
-The last sentence needs careful analysis to understand its meaning and it might not have meaning for anyone not familiar with the problem of overlapping intersections in a pattern.

This is tricky to solve succinctly, but here is an attempt that introduces the problem and then provides a solution:

"(Warning: overlapping intersections can cause problems in rendering the diagrams correctly to your specifications - Fix this by removing the overlapping intersection in the base pattern with a "-" key instruction in the input field so that every intersection occurs only once in the base pattern)"

The page is looking and working great otherwise :) Thank you for incorporating some of my other suggestions. I'm rather proud to have helped a little.

**Comment by jo-pol** [comment]
We are talking about an animated gif, which is something else than a youtube video. No control but way less storage. My concern also addresses small devices. Reloading and scrolling down makes you miss the first seconds. 

I just found a [trick](https://stackoverflow.com/questions/2573979/force-page-reload-with-html-anchors-html-js#2573986) to reload the page (and thus restart the animation) and have the animation in view. My idea is now to add a video control to the last frame and we would be all set.

For off-topic comments you can open a new issue, or a tiny pull request with additional questions.

**Comment by jo-pol** [comment]
I downloaded GIMP and tried to add something to the last frame like
![image](https://user-images.githubusercontent.com/10553630/100516790-9a5b8580-3186-11eb-9df9-91821930452e.png)
but have no luck to get the quality right, please help me to replace the [gif](https://github.com/d-bl/GroundForge/blob/master/docs/help/images/brick-to-overlap-animation.gif). Some (slightly blurred) transparency might be nice.

**Comment by veronika** [comment]
![brick-to-overlap-animation2](https://user-images.githubusercontent.com/259897/100519176-42ac2280-3164-11eb-8dd4-467f63c9a6aa.gif)

In Gimp, I opened the animation.  Each frame of the animation is layer.  I pasted the "player" image which created a "pasted layer".  I then converted this to an actual layer (context menu in layers tool set) .  I also created a layer with a solid white background, just before the "player" layer.

**Comment by jo-pol** [comment]
The restart trick is working, the animated image scrolls up as intended. Also applied further rewording and fine tuned some icons in the features section of the sidebar.

**Comment by jo-pol** [comment]
Celebrated too soon. I failed to notice it still opened on a new tab. Refreshing and staying on the same tab does not work.

**Comment by jo-pol** [comment]
This [revert](https://github.com/d-bl/GroundForge/commit/4ce91a541a2cfd9a32d41e2e2893034666883c07) shows the attempt. 

Browsers behaved different on the intended restart. Some stayed on the same tab and scrolled the animation up. One scrolled the animation too much up. One opened each restart in a new tab.

**Comment by jo-pol** [comment]
Another [trick](https://www.denisbouquet.com/replay-animated-gif-javascript/), would 0.3 MB be too big to download each time?

and [another](https://community.adobe.com/t5/dreamweaver/gif-play-on-hover-and-restart-after-hover-not-play-in-bg/td-p/7710797?page=1)

**Comment by jo-pol** [comment]
The last alternative seems to work. Please test with as many browser on as many devices as you can with https://jo-pol.github.io/GroundForge/help/Change-tiles

**Comment by jo-pol** [comment]
It was quite an elaborate detour. The recorded mp4 was too large, so I made an animated gif with a lot of drawbacks. Converting it back to mp4 reduce the size even more. So now we have all the standard video controls.

I came up with "alternating" as new tiling name. Any more suggestions?

**Comment by jo-pol** [comment]
fixed with #176
